### PR TITLE
nav: Adjust color to match PF4's new dark nav

### PR DESCRIPTION
**Pull request by garrett**
PatternFly 4 has changed to suggest a dark menu, similar to the one we had a while back. Usability studies were cited to show that people found the navigation to be clearer when it looked a bit separate with color swaps.

### CODE REVIEW

**Comment by garrett** [comment]
This would make Cockpit look more like the new navigation mockups currently @ https://garrett.github.io/cockpit-mockweb/navigation2/ and allows for us to be able to use a lighter header too, similar to the revised Firewall mockups.

You can see an HTML PatternFly 4 port of the Firewall mockup @ https://garrett.github.io/cockpit-mockweb/navigation2/network_firewall.html to see what it looks like in context. With the light navigation, the header looks odd. Therefore, this should land before the Firewall header changes.

**Comment by garrett** [comment]
TODO:

- [ ] Embolden the font weight for the selected item
- [ ] Adjust mobile (although it mostly works)
- [ ] Check in non-Firefox browsers
- [ ] Refine things a little further

**Comment by garrett** [comment]
Additional TODO:

- [ ] We'll want to adjust the search field too (#12579) either in this PR or that PR, depending on when things land.

**Comment by andreasn** [comment]
Here is the study itself. Probably good to reference in the commit.
https://blog.patternfly.org/patternfly/how-do-different-patternfly-visual-themes-affect-navigation-performance-perception-and-preference/
https://drive.google.com/file/d/1gPfp7GJ8MaImbgaivL8NzlNeucsXUBDw/view

**Comment by martinpitt** [comment]
I can't say I like it personally -- mixing dark and light themes gets you the worst of both worlds, a stressful high contrast at *both* day and night time. Therefore I'm surprised that the study said that this mix is supposed to lessen the eye strain..  But then again I keep being surprised by how many people use dark themes at daytime (which seems completely irrational to me).

But they *did* a study, which certainly beats my personal preference :grin: So either way, we certainly want to follow what PF4 does, so :+1: from me. There's still a TODO list, so not formally approving yet.

Thanks!

**Comment by garrett** [comment]
Yeah, I wanted to get it to a good point to share for initial feedback. I just need to check and double-check it and adjust it where necessary. Oh, and fix the font weight. (I'm not sure if we're pulling in all the weights we need. I'll have to investigate.)

**Comment by garrett** [comment]
I've been trying to cross-browser test all day, but have been hitting hurdle after hurdle.

I _know_ we have a bug in this PR with WebKit-based browsers, but:
- Any recent version of Epiphany (or any other WebKitGTK browser) is hitting https://bugs.webkit.org/show_bug.cgi?id=202321
- Ubuntu Stable has Inspector working, as its browser is quite old. However, it's so old, it's hitting the cert issue iOS and > 1 year old macOS has, so it's useless for testing Cockpit.
- macOS here has been giving me problem after problem after problem, as I haven't used it in forever and it's super-grumpy as a result. I got it to the point where I can do things, but then it refuses to even load Cockpit's login page over the network for some odd reason. (It's not the firewall. I had to fix that for Edge already.)

Sooooo this PR definitely has a bug, but I still can't properly debug it.

Meanwhile, I managed to test Chrome and it seems fine. Edge also was working for a minute or two until Windows hardlock crashed, after I fixed the firewall settings.

:scream:

Today must be technology-is-mad day.

**Comment by garrett** [comment]
Update: After rebooting the Mac _yet again_ and restarting Cockpit's socket _again_, I have Safari for macOS pointed at Cockpit. _This shouldn't have been so difficult._ :roll_eyes:

(Hopefully WebKit on non-Mac platforms gets unbroken again sometime soon.)

Cross-platform issues should now be resolved. Now to investigate the font.

**Comment by garrett** [comment]
Working on getting the search implemented like [the nav mockup](https://garrett.github.io/cockpit-mockweb/navigation2/). 

Here's the mockup:

Screenshot_2019-10-08 Cockpit


Here's a screenshot of in-browser changes:

Screenshot_2019-10-08 System - Rain

We have to make a bunch of changes to the code to fully overhaul the navigation (to move top-level into the navigation, move dashboard into the host switcher, support named groups, etc.), but it's starting to resemble the mockup.

The search icon is on the left to make it similar to the PF4 website and to gloss over the alignment of text, which just looks wrong when the search icon is on the right and the box aligns to the text. (That is, the text is noticably indented and looks incorrect.) Here's what _that_ looks like:

Screenshot_2019-10-08 System - Rain(1)

(Weird, right?)

I also tried aligning the text to the left (with the icon to the right) and it also feels wrong.

Screenshot_2019-10-08 Storage - Rain

Therefore, I think we should have the icon to the left, because it "feels" best and is also what the PF4 website does. In addition, the icons to the right are reserved for status icons, and this isn't really a status.

**Comment by andreasn** [comment]
Yeah, left alignment looks best.

**Comment by garrett** [comment]
I've rebased on #12579. The dark navigation now depends on search. I have the search PR as a commit in this one. We should land the search PR first (and the commit here will wash out).

File: pkg/shell/index.html [hunk 1]
```diff
@@ -360,7 +360,7 @@ <h4 class="modal-title" translatable="yes">Authentication</h4>
                     </div>
                 </div>
                 <div class="has-feedback search" id="menu-search">
-                    <input id="filter-menus" class="form-control" type="search" placeholder="Type to search..." translate="placeholder">
+                    <input id="filter-menus" class="form-control" type="search" placeholder="Search" translate="placeholder">
```
Comment (andreasn) [inline pkg/shell/index.html]: Axe gave me a critical error that this didn't have a label.

Comment (marusak) [reply pkg/shell/index.html]: > Axe gave me a critical error that this didn't have a label.

Hups. This is though not related to this PR as @garrett mentioned.

> Also, shouldn't these be translated?

It is translated, notice the `translate="placeholder"`

Comment (garrett) [reply pkg/shell/index.html]: Aha! Can translate also translate aria-label _and_ placeholder at the same time? (Does it accept more than one value, that is...)

File: node_modules/README [hunk 2]
```diff
@@ -1,9 +0,0 @@
-# Cockpit Javascript Dependencies
-
-For included Javascript libraries, and build processes
-Cockpit uses Node.js dependencies to process its code. The required Node.JS
-javascript code is packaged in the node_modules/ directory when distributed.
-
-Cockpit does not use Node.js or any npm packages code at runtime. In addition
-when building straight from a tarball it is also not necessary to have Node.js
-installed.
```
Comment (martinpitt) [inline node_modules/README]: This needs to be put back.

**Comment by martinpitt** [comment]
Current screenshot with "Search..." (The above still had "Type to search"):

image

**Pull request merged**